From c533aada209c517c1666989d7115672e7522aa72 Mon Sep 17 00:00:00 2001
From: Sylwester Lachiewicz <slachiewicz@apache.org>
Date: Sat, 13 Oct 2018 04:16:44 +0200
Subject: [PATCH 3/4] [MNG-6642] Revert "[MNG-5995] Remove dependency to
 maven-compat (#185)"

This partially reverts commit bb3ec5da71d26d105972392f0a20bc61bc5d8c53 to restore working with Tycho P2 repository.
---
 .../apache/maven/project/DefaultProjectBuildingHelper.java  | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuildingHelper.java b/maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuildingHelper.java
index 0bd94a86a..3159fac6b 100644
--- a/maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuildingHelper.java
+++ b/maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuildingHelper.java
@@ -33,7 +33,6 @@
 import org.apache.maven.artifact.Artifact;
 import org.apache.maven.artifact.InvalidRepositoryException;
 import org.apache.maven.artifact.repository.ArtifactRepository;
-import org.apache.maven.bridge.MavenRepositorySystem;
 import org.apache.maven.classrealm.ClassRealmManager;
 import org.apache.maven.model.Build;
 import org.apache.maven.model.Extension;
@@ -45,6 +44,7 @@
 import org.apache.maven.plugin.PluginManagerException;
 import org.apache.maven.plugin.PluginResolutionException;
 import org.apache.maven.plugin.version.PluginVersionResolutionException;
+import org.apache.maven.repository.RepositorySystem;
 import org.codehaus.plexus.PlexusContainer;
 import org.codehaus.plexus.classworlds.realm.ClassRealm;
 import org.codehaus.plexus.component.annotations.Component;
@@ -78,7 +78,7 @@
     private ProjectRealmCache projectRealmCache;
 
     @Requirement
-    private MavenRepositorySystem repositorySystem;
+    private RepositorySystem repositorySystem;
 
     @Requirement
     private MavenPluginManager pluginManager;
@@ -92,7 +92,7 @@
 
         for ( Repository repository : pomRepositories )
         {
-            internalRepositories.add( MavenRepositorySystem.buildArtifactRepository( repository ) );
+            internalRepositories.add( repositorySystem.buildArtifactRepository( repository ) );
         }
 
         repositorySystem.injectMirror( request.getRepositorySession(), internalRepositories );
-- 
2.21.0

